src/workloads/execution/SecondaryReadsGenny.yml

```diff
@@ -0,0 +1,252 @@
+        NopInPhasesUpTo: *max_phase
+        PhaseConfig:
+          Repeat: 1
+          Database: *db
+          Threads: 1
+          CollectionCount: 1
+          DocumentCount: *initial_nb_docs
+          BatchSize: *initial_nb_docs
+          Document: 
+            x: {^Inc: {}}
+            #s: {^Repeat: {count: *document_size, fromGenerator: {^Choose: {from: ["x"]}}}} --> why does this not work?
+            s: xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
+          Indexes:
+          - keys: {x: 1}
+
+- TemplateName: QuiesceTemplate
+  Config:
+    Name: {^Parameter: {Name: "Name", Default: Quiesce}}
+    Type: QuiesceActor
+    Threads: 1
+    Database: *db
+    Phases:
+      OnlyActiveInPhases:
+        Active: {^Parameter: {Name: "ActivePhases", Default: [-1]}}
+        NopInPhasesUpTo: *max_phase
+        PhaseConfig:
+          Repeat: 1
+            
+
+- TemplateName: WritesOnPrimaryTemplate
+  Config:
+    Name: {^Parameter: {Name: "Name", Default: WritesOnPrimary}}
+    Type: CrudActor
+    Threads: *writer_threads
+    Database: *db
+    Phases:
+      OnlyActiveInPhases:
+        Active: {^Parameter: {Name: "ActivePhases", Default: [-1]}}
+        NopInPhasesUpTo: *max_phase
+        PhaseConfig:
+          Duration: *test_duration
+          Collection: Collection0
+          Operations:
+          - OperationName: insertOne
+            OperationCommand:
+              Document:
+                x: {^RandomInt: {min: *initial_nb_docs, max: 1000000000}}
+                #s: {^Repeat: {count: *document_size, fromGenerator: {^Choose: {from: ["x"]}}}} --> why does this not work?
+                s: xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
+
+- TemplateName: ReadsOnSecondaryTemplate
+  Config:
+    Name: {^Parameter: {Name: "Name", Default: ReadsOnSecondary}}
+    Type: CrudActor
+    Threads: {^Parameter: {Name: "Threads", Default: -1}}
+    ClientName: Secondary
+    Database: *db
+    Phases:
+      OnlyActiveInPhases:
+        Active: {^Parameter: {Name: "ActivePhases", Default: [-1]}}
+        NopInPhasesUpTo: *max_phase
+        PhaseConfig:
+          Duration: *test_duration
```

Comment:
Should this maybe be phase_duration rather than test_duration? Open for argument there, but I might think of the whole run as the test, and that is going to be multiples of this value.

Reply:
Good point, phase_duration is closer to the truth than test_duration.